### PR TITLE
Acsf factory hooks suggestions

### LINKED ISSUE #2209: ACSF support for setup and sync
In 9.1.x, you can now create multiple databases locally for ACSF sites, BLT will automatically switch between them based on hostname and the multisites config array.

But we still need a way to run `blt setup` and `blt sync` against specific sites/databases. Seems like the easiest way is to automatically transform the site parameter into a local drush alias (e.g. `@project_name.site_name.local`). But this requires that local drush aliases exist for each site. We could probably automatically construct these in the same way that acsf-tools does via sfa for remote aliases.

Once this is set up, we should test and document the entire ACSF workflow.

### PR DESCRIPTION
**Pull request by lcatlett**
Fixes #2209 #2159 

Changes proposed:
- Adds post-theme-deploy hook to clear twig cache on all web heads on theme deploy
-Adds support for running blt commands on site install. This will allow for using config split and on site install for new sites.

### CODE REVIEW

**Comment by wimleers** [comment]
There should be a `@todo` to revert this at some point in the future, when all clients have a Drupal 8 version that is at least the one in which https://www.drupal.org/node/2752961 ships.

File: settings/acsf/post-install/post-install.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+$site = $_ENV['AH_SITE_GROUP'];
+$env = $_ENV['AH_SITE_ENVIRONMENT'];
+
+// The public domain name of the website.
+// Run updates against requested domain rather than acsf primary domain.
+$domain = $_SERVER['HTTP_HOST'];
+
+
+exec('/mnt/www/html/$target_env/vendor/acquia/blt/bin/blt deploy:update --define environment=$env --define drush.uri=$domain -v -y');
```
Comment (grasmash) [inline settings/acsf/post-install/post-install.php]: I don't think that variable substitution will work here since you're using single quotes.

Comment (grasmash) [reply settings/acsf/post-install/post-install.php]: Please use the long form of all options. E.g., `--verbose --yes`.

File: settings/acsf/post-install/post-install.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+/**
+ * @file
+ * Factory Hook: post-install.
+ *
+ * This hook enables you to execute PHP after a new website is created
+ * in your subscription. Unlike most API-based hooks, this hook does not
+ * take arguments, but instead executes the PHP code it is provided.
```
Comment (grasmash) [inline settings/acsf/post-install/post-install.php]: Add `@todo Remove this script after https://www.drupal.org/node/2752961 is resolved and the corresponding Drupal core version is set as a minimum requirement for BLT.`

File: settings/acsf/post-theme-deploy/clear-twig-cache.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# Clears TWIG cache for a site after a theme deploy. Executes via drush alias.
```
Comment (grasmash) [inline settings/acsf/post-theme-deploy/clear-twig-cache.sh]: Add @todo Remove this script after https://www.drupal.org/node/2752961 is resolved and the corresponding Drupal core version is set as a minimum requirement for BLT.

**Comment by damontgomery** [comment]
I opened #2440 to clear up some of the confusion around this PR.

**Pull request closed without merging**